include/envoy/local_info/local_info.h

```diff
@@ -25,17 +25,17 @@ class LocalInfo {
   /**
    * Human readable zone name. E.g., "us-east-1a".
    */
-  virtual const std::string& zoneName() const PURE;
+  virtual const std::string zoneName() const PURE;
```

Comment:
We could store local copy of the variable and return reference to it, i.e.

    const std::string& zoneName() const override {
      zone_name_ = node_.locality().zone();
      return zone_name_;
    }

but I'm not sure if that would result in any performance gains over the current change.

Since our `string != std::string`, the function returning `std::string&` returns reference to a temporary local variable (basically what I suggested above, but storing value into local variable and not a class member), which doesn't have any lifetime guarantees.